src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/storage/StaleBlobFinderTest.java

```diff
@@ -0,0 +1,116 @@
+import static org.mockito.Mockito.mock;
+import static uk.gov.hmcts.reform.bulkscanprocessor.util.TimeZones.EUROPE_LONDON_ZONE_ID;
+
+@ExtendWith(MockitoExtension.class)
+class StaleBlobFinderTest {
+
+    @Mock
+    private BlobServiceClient storageClient;
+
+    @Mock
+    private ContainerMappings containerMappings;
+
+    @InjectMocks
+    private StaleBlobFinder staleBlobFinder;
+
+    private OffsetDateTime expectedCreationTime = now().minus(3, ChronoUnit.HOURS);
+
+    @Test
+    void should_find_stale_blobs() {
+        given(containerMappings.getMappings())
+            .willReturn(
+                Arrays.asList(getMapping("bulkscan"), getMapping("cmc"), getMapping("sscs"))
+            );
+
+        var bulkscanBlobClient = mock(BlobContainerClient.class);
+        var cmcBlobClient = mock(BlobContainerClient.class);
+        var sscsBlobClient = mock(BlobContainerClient.class);
+        given(storageClient.getBlobContainerClient("bulkscan")).willReturn(bulkscanBlobClient);
+        given(storageClient.getBlobContainerClient("cmc")).willReturn(cmcBlobClient);
+        given(storageClient.getBlobContainerClient("sscs")).willReturn(sscsBlobClient);
+
+        mockStorageList(
+            bulkscanBlobClient,
+            Stream.of(
+                blob("bulk_scan_file_new", false),
+                blob("bulk_scan_file_stale", true)
+            )
+        );
+
+        mockStorageList(
+            cmcBlobClient,
+            Stream.of(
+                blob("cmc_scan_file_new_1", false),
+                blob("cmc_scan_file_new_2", false)
+            )
+        );
+
+        mockStorageList(sscsBlobClient, Stream.empty());
+
+        List<BlobInfo> blobInfos = staleBlobFinder.findStaleBlobs(2);
+
+        assertThat(blobInfos.size()).isEqualTo(1);
+        assertThat(blobInfos.get(0).container).isEqualTo("bulkscan");
+        assertThat(blobInfos.get(0).fileName).isEqualTo("bulk_scan_file_stale");
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss");
+        var expectedTime =
+            dateTimeFormatter.format(ZonedDateTime.ofInstant(expectedCreationTime.toInstant(), EUROPE_LONDON_ZONE_ID));
+
+        assertThat(blobInfos.get(0).createdAt).isEqualTo(expectedTime);
+    }
+
+    @SuppressWarnings("unchecked")
+    private void mockStorageList(BlobContainerClient blobClient, Stream<BlobItem> streamOfBlobItem) {
+        PagedIterable<BlobItem> listBlobsResult = mock(PagedIterable.class);
+        given(blobClient.listBlobs()).willReturn(listBlobsResult);
+        given(listBlobsResult.stream()).willReturn(streamOfBlobItem);
+
+    }
+
+    private BlobItem blob(String name, boolean staleFile) {
+        var blobItem = mock(BlobItem.class);
+        var properties = mock(BlobItemProperties.class);
+
+        given(blobItem.getProperties()).willReturn(properties);
+        if (staleFile) {
+            given(properties.getCreationTime()).willReturn(expectedCreationTime);
```

Comment:
how is this datetime 'expected'?
maybe just pass datetime to this method for blob creation instead of using external constant.